.github/workflows/test.yaml

```
@@ -346,3 +351,10 @@ jobs:
           KWOK_MODE: StableFeatureGateAndAPI
         run: |
           ./hack/e2e-test.sh kwokctl/kwokctl_${{ matrix.kwokctl-runtime }}
+
+      - name: Upload logs
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
```

```suggestion
        uses: actions/upload-artifact@v3
```

Maybe we should trust that it will remain compatible in v3